wagtail_footnotes/widgets.py

```diff
@@ -1,20 +1,64 @@
 from django.forms import HiddenInput
-from wagtail.utils.widgets import WidgetWithScript
+from wagtail import VERSION as WAGTAIL_VERSION
 
 
-class ReadonlyUUIDInput(WidgetWithScript, HiddenInput):
-    """
-    This isn't really read-only. It's a hidden input with an an adjacent div
-    showing the current value; that way we can set the value in JavaScript, but
-    the user can't easily change it.
-    """
+if WAGTAIL_VERSION and WAGTAIL_VERSION >= (6, 0):
+    from django.forms import Media
+    from django.utils.safestring import mark_safe
 
-    def render_html(self, name, value, attrs):
-        """Render the HTML (non-JS) portion of the field markup"""
-        hidden = super(WidgetWithScript, self).render(name, value, attrs)
-        display_value = value[:6] if value is not None else value
-        shown = f'<div id="{attrs["id"]}_display-value" style="padding-top: 1.2em;">{display_value}</div>'
-        return shown + hidden
+    class ReadonlyUUIDInput(HiddenInput):
+        """
+        This isn't really read-only. It's a hidden input with an an adjacent div
+        showing the current value; that way we can set the value in JavaScript, but
+        the user can't easily change it.
+        """
 
-    def render_js_init(self, id_, name, value):
-        return f'setUUID("{id_}");'
+        def render(self, name, value, attrs=None, renderer=None):
+            # no point trying to come up with sensible semantics for when 'id' is missing from attrs,
+            # so let's make sure it fails early in the process
+            try:
+                attrs["id"]
+            except (KeyError, TypeError) as exc:
+                raise TypeError(
+                    "ReadonlyUUIDInput cannot be rendered without an 'id' attribute"
+                ) from exc
+
+            widget_html = self.render_html(name, value, attrs)
+
+            return mark_safe(widget_html)  # noqa: S308
```

Comment:
This section was copied from Wagtail's `WidgetWithScript`:
https://github.com/wagtail/wagtail/blob/main/wagtail/utils/widgets.py#L27